javascript/javascript2/week2/main.js

```diff
@@ -0,0 +1,129 @@
+// Doubling of number
+
+const numbers = [1, 2, 3, 4];
+const newNumbers = numbers
+  .filter((x) => x % 2 !== 0)
+  .map((filteredNumber) => filteredNumber * 2);
+console.log(newNumbers);
+
+// Create an array  containing the movies with a short title
+
+const shortMoviesTitle = movies.filter((movie) => !movie.title.includes(" "));
+console.log(shortMoviesTitle);
+
+// Create an array of movie titles with long movie titles
+
+const longMoviesTitles = movies
+  .filter((movie) => movie.title.includes(" "))
+  .map((filteredMovie) => filteredMovie.title);
+console.log(longMoviesTitles);
+
+// Count the number of movies made between 1980-1989 (including both the years)
+
+const numberOfMoviesMadeBetweenYears = movies.reduce((movieCounter, movie) => {
+  if (movie.year >= 1980 && movie.year <= 1989) {
+    return (movieCounter += 1);
+  }
+  return movieCounter;
+}, 0);
```

Comment:
POWERFUL! Excellent use of `.reduce`. This is the hardest method to grok, well done!